File: ev-frontend/src/app/models/charging-station.ts
export interface ChargingStation {
  id: string;
  ocmUuid: string;
  operatorName?: string;
  latitude: number;
  longitude: number;
  postcode: string;
  londonBorough: string;
  totalConnectors: number;
  availableConnectors: number;
  currentStatus: 'AVAILABLE' | 'OCCUPIED' | 'FAULTY' | 'UNKNOWN';
  lastUpdated: string;
}

export interface StatusChangeEvent {
  stationUuid: string;
  oldStatus: string;
  newStatus: string;
  londonBorough: string;
  availableConnectors: number;
  totalConnectors: number;
  changedAt: string;
}

File: ev-backend/src/main/java/com/evmonitor/model/ChargingStation.java
package com.evmonitor.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.Instant;
import java.util.UUID;

@Data
@Entity
@Table(name = "charging_stations")
public class ChargingStation {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "ocm_uuid", unique = true)
    private String ocmUuid;

    private String operatorName;
    private Double latitude;
    private Double longitude;
    private String postcode;
    private String londonBorough;
    private Integer totalConnectors;
    private Integer availableConnectors;

    @Column(name = "current_status")
    private String currentStatus = "UNKNOWN";

    @Column(name = "last_updated")
    private Instant lastUpdated = Instant.now();
}